packages/swing-store/test/test-state.js

```diff
@@ -120,114 +147,130 @@ test('persistent kvStore maxKeySize write', async t => {
-  s1pos = streamStore.writeStreamItem('st1', 'fourth', s1pos);
-  s2pos = streamStore.writeStreamItem('st2', 'twoth', s2pos);
-  const s2posAlt = s2pos;
-  s2pos = streamStore.writeStreamItem('st2', 'threeth', s2pos);
-  s2pos = streamStore.writeStreamItem('st2', 'fourst', s2pos);
-  streamStore.closeStream('st1');
-  streamStore.closeStream('st2');
-  const reader1 = streamStore.readStream('st1', start, s1pos);
-  t.deepEqual(Array.from(reader1), ['first', 'second', 'third', 'fourth']);
-  s2pos = streamStore.writeStreamItem('st2', 're3', s2posAlt);
-  streamStore.closeStream('st2');
-  const reader2 = streamStore.readStream('st2', start, s2pos);
-  t.deepEqual(Array.from(reader2), ['oneth', 'twoth', 're3']);
-
-  const reader1alt = streamStore.readStream('st1', s1posAlt, s1pos);
+  transcriptStore.initTranscript('st1');
+  transcriptStore.initTranscript('st2');
+  transcriptStore.writeTranscriptItem('st1', 'first');
+  transcriptStore.writeTranscriptItem('st1', 'second');
+  transcriptStore.rolloverSpan('st1');
+  transcriptStore.writeTranscriptItem('st1', 'third');
+  transcriptStore.writeTranscriptItem('st2', 'oneth');
+  transcriptStore.writeTranscriptItem('st1', 'fourth');
+  transcriptStore.writeTranscriptItem('st2', 'twoth');
+  transcriptStore.writeTranscriptItem('st2', 'threeth');
+  transcriptStore.writeTranscriptItem('st2', 'fourst');
+  transcriptStore.closeTranscript('st1');
+  transcriptStore.closeTranscript('st2');
+  const reader1 = transcriptStore.readTranscript('st1', 0);
+  t.deepEqual(Array.from(reader1), ['first', 'second']);
+  const reader2 = transcriptStore.readTranscript('st2', 0);
+  t.deepEqual(Array.from(reader2), ['oneth', 'twoth', 'threeth', 'fourst']);
+
+  t.throws(() => transcriptStore.readTranscript('st2', 3), {
+    message: 'no transcript for st2 at 3',
+  });
+
+  const reader1alt = transcriptStore.readTranscript('st1');
   t.deepEqual(Array.from(reader1alt), ['third', 'fourth']);
+  const reader1alt2 = transcriptStore.readTranscript('st1', 2);
+  t.deepEqual(Array.from(reader1alt2), ['third', 'fourth']);
 
-  const emptyPos = streamStore.writeStreamItem('empty', 'filler', start);
-  streamStore.closeStream('empty');
-  const readerEmpty = streamStore.readStream('empty', emptyPos, emptyPos);
+  transcriptStore.initTranscript('empty');
+  transcriptStore.closeTranscript('empty');
+  const readerEmpty = transcriptStore.readTranscript('empty');
   t.deepEqual(Array.from(readerEmpty), []);
-  const readerEmpty2 = streamStore.readStream('empty', start, start);
-  t.deepEqual(Array.from(readerEmpty2), []);
+
+  t.throws(() => transcriptStore.readTranscript('nonexistent'), {
+    message: 'no current transcript for nonexistent',
+  });
 
   await commit();
+  t.deepEqual(exportLog.getLog(), [
+    [
+      [
+        'export.transcript.st1.0',
+        '{"vatID":"st1","hash":"d3face142175f5507cf261be08988b674aa622b4ce8d958a0b3c740592f1113e","startPos":0,"endPos":2,"size":13}',
```

Comment:
Can we compute this from the expected content instead of hard coding the hash?

Reply:
I'd have to double check, but I think this test was intended to alert us to the fact that we were changing a schema (and thus breaking compatibility with existing deployments), so we wouldn't do that accidentally.

(that doesn't necessarily mean we shouldn't be doing a manual hash computation, which would serve as a "does it do what our specification claims" check)

Reply:
I see this wasn't addressed but it was marked as resolved. If there was a decision to not do this, can it be recorded?